src/content/effective-dart/usage.md

```diff
@@ -325,7 +326,7 @@ Of course, if `null` is a valid initialized value for the variable,
 then it probably does make sense to have a separate boolean field.
 
 
-### CONSIDER assigning a nullable field to a local variable to enable type promotion
+### CONSIDER work arounds to enable type promotion for nullable fields
```

Comment:
Grammar nit: I think "workaround" is a single word when used as a noun.

More generally, I think the naming of this guideline makes less sense when it includes the null-check pattern because that technically isn't using type promotion. Instead of promoting the type of the variable after it's declared, it binds an entirely new variable whose static type is already the non-nullable base type.

Maybe it's better to split this into two separate guidelines? Something like:

---

**CONSIDER using a null-check pattern to work with a nullable expression**

*null-check pattern example...*

**CONSIDER assigning a nullable expression to a local variable to enable type promotion**

*other example...*

---

I'm saying "expression" instead of "field" here because these patterns are equally applicable to more complex expressions that return nullable values, like:

```dart
if (foo.bar.baz.someString != null) {
  print(foo.bar.baz.someString.length); // Error, can't promote to non-nullable.
}

if (foo.bar.baz.someString case var someString?) {
  print(someString.length); // Works.
}

var someString = foo.bar.baz.someString;
if (someString != null) {
  print(someString.length); // Also works.
}
```

Reply:
I did see your suggestion in the issue to split them into two. The reason I kept them together is because both the situation and the outcome are identical. Like, both the local variable and the pattern methods are just two different ways to achieve the same thing when you have the same problem, if that makes sense, and it felt natural to group them based on that common purpose. 

Another title I thought of was "**Consider workarounds to bypass type promotion limitations**" -- is that general enough to keep both strategies together under one heading? 

Or, is the issue _because_ of the generic title, and Effective Dart headings need to be more specific than that? If that's the case then I will split them up.

Reply:
> I did see your suggestion in the issue to split them into two.

Oh, heh, I completely forgot I already suggested that. At least I'm consistent with my past self. :)

> The reason I kept them together is because both the situation and the outcome are identical. Like, both the local variable and the pattern methods are just two different ways to achieve the same thing when you have the same problem

SGTM.

> Another title I thought of was "**Consider workarounds to bypass type promotion limitations**" -- is that general enough to keep both strategies together under one heading?

It's not really about type promotion itself, because type promotion is another solution. The problem is really that you can't do much with an expression whose type is nullable. It's about escaping the nullable type shackles.

Maybe we can be more concrete and just mention both solutions in the guideline? Something like "CONSIDER type promotion or null-check patterns for using nullable types"?

Reply:
> CONSIDER type promotion or null-check patterns for using nullable types

That sounds great! Thanks for working something out with me. Your consistency is appreciated :)